### ISSUE TITLE
Missing keyboard/screen reader support

### ISSUE THREAD

**Issue by it-vegard** [DavidRnR/ngx-checkbox#1]
Hi,

Tested on: Chrome 66.0 on Windows 10 (but should apply everywhere)

This checkbox does not seem to support keyboard navigation, and thus not screen readers either, making it in violation with [WCAG 2.0 success criteria 2.1.1](https://www.w3.org/TR/WCAG20/#keyboard-operation).

Reason:
The input has "display: 'none'", which removes it from the tab order. This makes it inaccessible to keyboard and screen reader users. It also seems to lack focus styling.

Proposed solution:
You either need to:

1. just visually hide it (e.g. moving it off-screen) or 
2. add "tabindex='0'" to some other element that will listen to "space"-keypresses. 

If using option 2, you also have  to remember to add "role='checkbox'", "aria-checked='true/false'" (remember to update) and make sure the label will be read out as well.

I'd go with option 1 (moving the input off-screen), as that will give you a lot of accessibility-functionality for free, since keyboard users will interact with the real checkbox. ("position='absolute';left: -9999em" should do the trick) 

Also remember to add focus styling to the visual checkbox!

**Comment by DavidRnR** [comment]
Thanks for the feedback @it-vegard . I really appreciate it.

Version 0.0.6 is available now:

New features:
- tabindex - @Input()
- id - @Input() to related with a label
- aria-checked
- role="checkbox"
- focus style
- keypressed by space

Thanks!

**Comment by it-vegard** [comment]
Great! Looks much better now.

But the label still does not seem to work properly. I think it is because the "for"-attribute references the real checkbox, while that has display="none" and will not be interacted with directly. The solution should probably be to place the id on the span with role="checkbox" instead, although i 

It could alternatively have something to do with having the custom checkbox wrapped in a label as well as having an "external" label referencing it through "for"-attribute. I don't know exactly how that works in different browsers/screen readers, but it could possibly make some problems.
